src/ol/renderer/canvas/intermediatecanvas.js

```diff
@@ -45,7 +45,8 @@ ol.renderer.canvas.IntermediateCanvas.prototype.composeFrame = function(frameSta
 
     // clipped rendering if layer extent is set
     var extent = layerState.extent;
-    var clipped = extent !== undefined;
+    var clipped = extent !== undefined &&
+        !ol.extent.containsExtent(extent, frameState.extent);
```

Comment:
We need to clip when the layer extent intersects but does not contain the frame state's extent. This check only does the 2nd part of the logic.